source/math/FloatVectorList.ooc

```diff
@@ -381,4 +381,80 @@ FloatVectorList: class extends VectorList<Float> {
 			result add(this[i] clamp(floor, ceiling))
 		result
 	}
+	findMaxIndex: func -> (Int, Float) {
+		index := 0
+		for (i in 1 .. this count)
+			if (this[index] < this[i])
+				index = i
+		(index, this[index])
+	}
+	crossCorrelation: func (other: This, vectorLength: Int) -> Float {
+		upperPart := 0.0f
+		lowerPart := 0.0f
+		lowerPart2 := 0.0f
+		for (i in 0 .. vectorLength) {
+			upperPart += ((this[i] - this mean) * (other[i] - other mean))
```

Comment:
`this[i]` causes an expensive memcpy operation. Before starting the loop, create the following variables:
```ooc
thisPointer := (this pointer as Float*)
otherPointer := (other pointer as Float*)
```
and access those instead:
`(thisPointer[i] - thisMean) * (otherPointer[i] - otherMean)`
etc. This should improve your performance enough to make your printouts below unnecessary.